docs/training/apis-howto/api-core-ug.rst

```diff
@@ -1,212 +1,302 @@
 
-By checkpointing periodically during training and reporting those checkpoints to the master, you can
-stop and restart training in two different ways: either by pausing and reactivating training using
-the WebUI, or by clicking the Continue Trial button after the experiment completes.
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_metrics.py
+   :language: python
+   :start-after: # Docs snippet start: report training metrics
+   :end-before: # Docs snippet end: report training metrics
+   :dedent:
 
-These two types of continuations have different behaviors. While you always want to preserve the
-value you are incrementing (the "model weight"), you do not always want to preserve the batch index.
-When you pause and reactivate you want training to continue from the same batch index, but when
-starting a fresh experiment you want training to start with a fresh batch index. You can save the
-trial ID in the checkpoint and use it to distinguish the two types of continues.
+and ``core_context.train.report_validation_metrics()``:
 
-#. Create a new ``2_checkpoints.py`` training script called by copying the ``1_metrics.py`` script
-   from :ref:`core-metrics`.
+.. literalinclude:: ../../../examples/tutorials/core_api_pytorch_mnist/model_def_metrics.py
+   :language: python
+   :start-after: # Docs snippet start: report validation metrics
+   :end-before: # Docs snippet end: report validation metrics
+   :dedent:
 
-#. Write save and load methods for your model:
+.. note::
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/2_checkpoints.py
-      :language: python
-      :pyobject: save_state
+   Viewing Epoch-Based Metrics in the WebUI
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/2_checkpoints.py
-      :language: python
-      :pyobject: load_state
+   You can modify your code to report epoch-based metrics as shown in this example. When you report
+   epoch-based metrics during training, you'll be able to view epoch-based metric data in the WebUI.
 
-#. In your ``if __name__ == "__main__"`` block, use the ClusterInfo API to gather additional
-   information about the task running on the cluster, specifically a checkpoint to load from and the
-   trial ID, which you also pass to ``main()``.
+.. tip::
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/2_checkpoints.py
-      :language: python
-      :start-at: info = det.get_cluster_info()
+   Avoiding a Duplicate Key Error
 
-   It is recommended that you always follow this pattern of extracting values from the ClusterInfo
-   API and passing the values to lower layers of your code, instead of accessing the ClusterInfo API
-   directly in the lower layers. In this way the lower layer can be written to run on or off of the
-   Determined cluster.
+   It is best to stitch metrics together in one continuous graph as your experiment progresses. To
```

Comment:
The instructions here are pretty unclear, IMO.

Seems like a Warning is more appropriate than a Tip, also: Metrics can only be reported once per `steps_completed` value, otherwise a duplicate key error will be raised.

(I don't recall if you're allowed to report training and validation metrics with the same `steps_completed`, or if the restriction only applies to calling one of those twice)

Reply:
honestly, we're working on unified metrics where this won't be a problem.

So I'd vote for just eliminating this warning, but I don't feel too strongly about it.

If we keep it, @garrett361 is right about the limitations.